vehicle/DieselEngine.h

```diff
@@ -0,0 +1,22 @@
+#ifndef DIESEL_ENGINE_H
+#define DIESEL_ENGINE_H
+
+#include "IEngine.h"
+
+class DieselEngine : public Engine {
+public:
+    DieselEngine(int cylinders) : cylinders(cylinders) {}
+
+    std::string getFuelType() const override {
+        return "Diesel";
+    }
+
+    int getNumberOfCylinders() const override {
+        return cylinders;
+    }
+
+private:
+    int cylinders;
```

Comment:
use some name convention: give some prefix to class member names

Reply:
names changed